components/funder/src/mutual_credit/outgoing.rs

```diff
@@ -191,7 +191,8 @@ impl OutgoingMc {
         let response_signature_buffer =
             create_response_signature_buffer(&response_send_funds, &pending_transaction);
         // The response was signed by the destination node:
-        let dest_public_key = pending_transaction.route.public_keys.last().unwrap();
+        // Note the key order is reversed from our perspective
+        let dest_public_key = &self.mutual_credit.state().idents.local_public_key;
```

Comment:
Is it, actually?

Reply:
This is somewhat similar to the incoming.rs issue of finding the signing key, but there is a slight difference.
As this is the outgoing case, we might be the destination of the request, sending back a signed response. In this case the route will be empty, and we won't be able to get the last element safely.

What I suggest is to have an if statement. If we do manage to get the last element, this is the `dest_public_key`. Otherwise, the `dest_public_key` should be the `local_public_key` (Because, we are the destination in that case).

Please tell me if it works for you!

Reply:
It seems to work. Thanks